includes/Checker/Checks/Abstract_PHP_CodeSniffer_Check.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Checks\PHP_CodeSniffer_Check
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker\Checks;
+
+use WordPress\Plugin_Check\Checker\Check;
+use WordPress\Plugin_Check\Checker\Check_Result;
+use Exception;
+
+/**
+ * Check for running one or more PHP CodeSniffer sniffs.
+ *
+ * @since n.e.x.t
+ */
+abstract class Abstract_PHP_CodeSniffer_Check implements Check {
+
+	/**
+	 * List of allowed PHPCS arguments.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @var array
+	 */
+	protected $allowed_args = array(
+		'extensions',
+		'standard',
+		'sniffs',
+		'exclude',
+	);
+
+	/**
+	 * Returns an associative array of arguements to pass to PHPCS.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array
+	 */
+	abstract public function get_args();
+
+	/**
+	 * Creates the command arguments.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check results including the plugin context to check.
+	 *
+	 * @return array
+	 */
+	private function create_args( Check_Result $result ) {
```

Comment:
Instead of calling `$this->get_args()` in this method, how about we make it a bit more "standalone", so that the value of `$this->get_args()` is passed to the method as a parameter? This way it becomes less coupled to that concrete value. A bit of a nit-pick, but makes the code cleaner and more reusable.

If we do that, maybe we also want to rename this method to something like `parse_args` or `parse_argv` to clarify the purpose better?